src/common/config/index.ts

```diff
@@ -552,101 +455,52 @@ export class Config extends EventEmitter {
+        } else {
+            const dir = path.dirname(filePath);
+            if (!fs.existsSync(dir)) {
+                fs.mkdirSync(dir);
+            }
 
-        const config = {...this.combinedData};
-        if (option) {
-            return config[option];
+            fs.writeFileSync(filePath, json, 'utf8');
         }
-        return config;
     }
 
-    handleGetLocalConfiguration = (event: Electron.IpcMainInvokeEvent, option: keyof ConfigType) => {
-        log.debug('Config.handleGetLocalConfiguration', option);
-
-        const config: Partial<LocalConfiguration> = {...this.localConfigData};
-        config.appName = app.name;
-        config.enableServerManagement = this.combinedData?.enableServerManagement;
-        config.canUpgrade = this.canUpgrade;
-        if (option) {
-            return config[option];
+    private checkWriteableApp = () => {
+        if (!this.appPath) {
+            throw new Error('Config not initialized, cannot regenerate');
         }
-        return config;
-    }
-
-    handleUpdateTeams = (event: Electron.IpcMainInvokeEvent, newTeams: TeamWithTabs[]) => {
-        log.debug('Config.handleUpdateTeams');
-        log.silly('Config.handleUpdateTeams', newTeams);
-
-        this.set('teams', newTeams);
-        return this.combinedData!.teams;
-    }
 
-    /**
-     * Detects changes in darkmode if it is windows or osx, updates the config and propagates the changes
-     * @emits 'darkModeChange'
-     */
-    handleUpdateTheme = () => {
-        log.debug('Config.handleUpdateTheme');
+        if (process.platform === 'win32') {
+            try {
+                fs.accessSync(path.join(path.dirname(this.appPath), '../../'), fs.constants.W_OK);
 
-        if (this.combinedData && this.combinedData.darkMode !== nativeTheme.shouldUseDarkColors) {
-            this.combinedData.darkMode = nativeTheme.shouldUseDarkColors;
-            this.emit('darkModeChange', this.combinedData.darkMode);
-        }
-    }
+                // check to make sure that app-update.yml exists
+                if (!fs.existsSync(path.join(process.resourcesPath, 'app-update.yml'))) {
+                    log.warn('app-update.yml does not exist, disabling auto-updates');
+                    return false;
+                }
+            } catch (error) {
+                log.info(`${this.appPath}: ${error}`);
+                log.warn('autoupgrade disabled');
+                return false;
+            }
 
-    /**
-     * Manually toggles dark mode for OSes that don't have a native dark mode setting
-     * @emits 'darkModeChange'
-     */
-    toggleDarkModeManually = () => {
-        if (!this.combinedData) {
-            return;
+            // eslint-disable-next-line no-undef
+            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+            // @ts-ignore
+            return __CAN_UPGRADE__; // prevent showing the option if the path is not writeable, like in a managed environment.
```

Comment:
Not something for this PR, but you can define a type for this in a `.d.ts` file to make it available everywhere without having to fight with TypeScript

Reply:
Ahh right, I should probably do that.